### ISSUE TITLE
One-hot labels produce dim mismatch

### ISSUE THREAD

**Issue by castagninojose** [castagninojose/imgalaxy#18]
Current state of the codebase produces an error when selecting one-hot encoded labels, i.e., when `sparse=False`. 

Using such labels would enable the use of other loss functions, which would in turn enlarge the hyperparameter space we can explore in the second stage of training.

Goal: Troubleshoot this bug to enable the use of either sparse or one-hot labels.

**Comment by castagninojose** [comment]
There's dedicated one-hot versions of IoU metric. Using these fixes this.

Will be closed by #17

**Comment by castagninojose** [comment]
Fixed in #17
